### ISSUE TITLE
Better error messages if geolocation fails due to the usage of http instead of https on Google Chrome?

### ISSUE THREAD

**Issue by seyfro** [domoritz/leaflet-locatecontrol#160]
As you surely know, with Google Chrome 50+ geolocation is only allowed from the browser side anymore if the site is delivered security via https, showing the following popup:

![geolocation-error-info](https://cloud.githubusercontent.com/assets/158669/14827067/e9edebea-0be0-11e6-8c0c-a5222e23bbe6.png)

I wonder if there is a better way to tell users that this is not due to a bug in locatecontrol? Anyway detecting Chrome browsers via JS might not be very exact & I am pretty sure that other browser manufactor will follow soon with similar security/privacy enhancements.

For now I set up an info page for my plugin users at https://www.mapsmarker.com/faqs/geolocaton-only-google/ - anyway I currently do not see a way to display a link to that url instead of the (very technical) Google info-page https://goo.gl/Y0ZkNV

**Comment by domoritz** [comment]
Yeah, I noticed that on my demo page and took it as a reason to upgrade my website to https. I don't see a good way to handle this from the plugin side. The most important thing would be to notify developers. I should probably put a note in the readme.

**Comment by ghost** [comment]
Chrome will not support geolocation with HTTP anymore, and there will be no way around it.

The message should be handled on the app side, not the plugin. `onLocationError` can be used for this just perfectly - it will override the default popup message.

**Comment by wsenjer** [comment]
Thanks - using onLocationError seems a good aproach. Anyway: does anyone know if the default Chrome alert window can be hidden? If we add our own location error message, the default Chrome alert still shows up after our own location alert is closed.

**Comment by ghost** [comment]
Does this work for you?

Default behaviour:
http://jsfiddle.net/z49r7jz9/

With `onLocationError`:
http://jsfiddle.net/L1wp5fcs/

**Comment by seyfro** [comment]
yes thx - this works for us (start geolocation on http://www.mapsmarker.com/dev/pro/2016/04/03/v2-7-preview/ vs https://www.mapsmarker.com/dev/pro/2016/04/03/v2-7-preview/ for a custom error alert)

**Comment by seyfro** [comment]
I decided to follow this approach for Maps Marker Pro: set an empty onLocationError function according to your example in the constructor which gets added if visitor is using chrome (globale $is_chrome) and on a non-https-site (!is_ssl()). In addition I added an additional event handler for the same visitor showing a custom error message. So locationerrors are only suppressed for Chrome & http and not globally for everyone.

thx for your help - closing ticket....
